js/style/style.js

```diff
@@ -247,19 +251,95 @@ Style.prototype = util.inherit(Evented, {
         zh.lastZoom = z;
     },
 
+    _checkLoaded: function () {
+        if (!this._loaded) {
+            throw new Error('Style is not done loading');
+        }
+    },
+
     /**
-     * Apply multiple style mutations in a batch
-     * @param {function} work Function which accepts the StyleBatch interface
+     * Apply queued style mutations in a batch
      * @private
      */
-    batch: function(work) {
-        styleBatch(this, work);
+    update: function(classes, options) {
```

Comment:
I'd also leave `reloadSource` -> `updateSource` rename for another PR because it's tricky, potentially involving a lot of changes that are out of scope for this PR (e.g. it calls `source.reload`, which in turn calls `pyramid.reload`, but `pyramid` also has `update` method which is different, so you also have to rethink pyramid API).